lightning/src/util/events.rs

```diff
@@ -199,6 +208,30 @@ pub enum Event {
 		/// payment is to pay an invoice or to send a spontaneous payment.
 		purpose: PaymentPurpose,
 	},
+	/// Indicates a payment has been claimed and we've received money!
+	///
+	/// This most likely occurs when [`ChannelManager::claim_funds`] has been called in response
+	/// to an [`Event::PaymentReceived`]. However, if we previously crashed during a
+	/// [`ChannelManager::claim_funds`] call you may see this event without a corresponding
+	/// [`Event::PaymentReceived`] event.
+	///
+	/// # Note
+	/// LDK will not stop an inbound payment from being paid multiple times, so multiple
+	/// `PaymentReceived` events may be generated for the same payment. If you then call
+	/// [`ChannelManager::claim_funds`] twice for the same [`Event::PaymentReceived`] you may get
+	/// multiple `PaymentClaimed` events.
+	///
+	/// [`ChannelManager::claim_funds`]: crate::ln::channelmanager::ChannelManager::claim_funds
+	PaymentClaimed {
+		/// The hash for which the preimage should be handed to the ChannelManager. Note that LDK will
+		/// not stop you from registering duplicate payment hashes for inbound payments.
+		payment_hash: PaymentHash,
+		/// The value, in thousandths of a satoshi, that this payment is for.
+		amt: u64,
+		/// Information for claiming this received payment, based on whether the purpose of the
+		/// payment is to pay an invoice or to send a spontaneous payment.
```

Comment:
Replace with: "The purpose of this claimed payment, i.e. whether the purpose of the payment was to pay an invoice or to send a spontaneous payment."